cirq-google/cirq_google/ops/z_gate_family.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+import cirq
+from cirq.ops import raw_types
+from cirq_google.ops.physical_z_tag import PhysicalZTag
+
+
+_PHYSICAL_Z_TAG = PhysicalZTag()
+
+
+class ZGateFamily(cirq.GateFamily):
+    """A GateFamily which accepts either virtual or physical Z gates from Google devices.
+
+    If `physical_z` is true, accepts physical Z gates, i.e. `TaggedOperations` containing a Z gate
+    and a `cirq_google.PhysicalZTag`.
+    Otherwise, Accepts virtual Z gates, i.e. Z gates without a `cirq_google.PhysicalZTag`.
+    """
+
+    def __init__(self, physical_z=False) -> None:
+        self._physical_z = physical_z
+        super().__init__(cirq.Z, ignore_global_phase=True)
```

Comment:
Are you sure this is what you want?  The device can accept any angle (ie. ZPowGate), not just the Pauli Z gate.

Reply:
This will not be enough and we will need to add ZPowGate and not just Pauli Z. 

Also, see my comment above not having the `ZGateFamily` at all.

Reply:
Ack, misunderstanding on my part